packages/react/src/SecureRoute.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { FC, useEffect, useRef } from 'react';
+import { Route, RouteProps } from 'react-router-dom';
+
+import useLogto from './use-logto';
+
+const SecureRoute: FC<RouteProps> = ({ ...routeProperties }) => {
+  const { isAuthenticated, logtoClient, isLoading } = useLogto();
+  // The following useEffect takes several deps,
+  // make sure only happen when isAuthenticated is changed,
+  // prevent calling multiple times
+  const pendingLoginReference = useRef(false);
```

Comment:
don't know if `useRef()` here is the right choice. it's more like a state from the code below.

Reply:
it's okay to call the `useEffect` multiple times, as long as we can ensure the right one (with `login()` called) will be just called once when all criteria are satisfied.

Reply:
use `pendingLoginReference` as a `lock` for async function `login`, to ensure there is only one excecuting `login`, why useRef instead of useState?
1. will NOT cause re-render or call `useEffect`, avoid loop.
2. not be used in `render`

Reply:
outdated, thouse code has been removed